service/controller/v1/resource/drainer/create.go

```diff
@@ -144,6 +141,19 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "no pods to be deleted running system workloads", "clusterID", clusterID)
 	}
 
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting guest cluster node from Kubernetes API")
+
+		n := key.NodeNameFromDrainerConfig(drainerConfig)
+		o := &apismetav1.DeleteOptions{}
+		err := k8sClient.CoreV1().Nodes().Delete(n, o)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleted guest cluster node from Kubernetes API")
+	}
```

Comment:
This should be all the magic. In case a node is not there anymore we catch the case above and cancel early.

Reply:
I am unsure again here: We delete the pods above, does it matter if the pods aren't properly deleted yet when we delete the node? 
What happens if pods get stuck during deletion? What do we do right now when removing a node?

I feel like I am missing quite a bit of the picture here, might just be me.

Reply:
Deleting a node is basically deregistering it. AFAIK this is not about fucking with the workloads. Would be good to know what exactly happens to workloads though but here again, the draining is implemented like that already and the provider operators rip the guest cluster nodes right at this point anyway. We need more automated testing for this and decide based on data.

Reply:
Ack, I wasn't aware that provider operators do exactly this.

Reply:
AFAIUI the draining was already done. If pods are stuck in termination I'd say it's even good we forcefully delete them (if that is happening).

Reply:
AFAIK we use Evict API for deleting a pod and this means that the request is finished after pod is removed so no problem,s its basically same behavior as `kubectl drain`

Reply:
Pods are deleted like:
```
err := k8sClient.CoreV1().Pods(p.GetNamespace()).Delete(p.GetName(), &apismetav1.DeleteOptions{})
```

Is it evict API? I think it's pods/delete.